### ISSUE TITLE
brunch of release3.6 build never finish on VS2017

### ISSUE THREAD

**Issue by KevinYeti** [craftworkgames/MonoGame.Extended#597]
I just download the brunch of release3.6 and start a building of the demo project Platformer.
But the building process seems never finish . When I looked into the process list of the task manager ,  a msbuild.exe process is running. I 've waited it for two hours and nothing happens. 

So is there any suggestion for that ?

BTW, the MGCB.exe did not work well on my PC. I guess the content build processing blocked the compiling but not sure.

**Comment by craftworkgames** [comment]
It should only take 3 or 4 minutes to do the build. So no need to wait for hours. Here's the output from a recent build on our build server.

```
[03:21:23][Step 1/1] Task                          Duration            
[03:21:23][Step 1/1] --------------------------------------------------
[03:21:23][Step 1/1] Restore                       00:00:11.2892650    
[03:21:23][Step 1/1] Build                         00:00:47.4824236    
[03:21:23][Step 1/1] Test                          00:01:14.7740188    
[03:21:23][Step 1/1] Pack                          00:01:05.8427525    
[03:21:23][Step 1/1] --------------------------------------------------
[03:21:23][Step 1/1] Total:                        00:03:19.3886408    
```

As for the problem you're experiencing I'm not sure. I've never seen that problem before. My only suggestion would be to make sure you're running the latest versions:
 - [Visual Studio](https://visualstudio.microsoft.com/downloads/) (community edition works fine)
 - [MonoGame](http://www.monogame.net/downloads/) (I run the latest development version myself) 
 - and the [.NET Core SDK](https://dotnet.microsoft.com/download).

Have you tried running the `build.ps1` script from the command line? This is how the build system does it.

**Comment by KevinYeti** [comment]
update：
After waiting a long time , I finially got this error:
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(45,5): error MSB4018: “GetMGCBTask” Task fail。
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(45,5): error MSB4018: System.Net.WebException:  Operation timeout。
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(45,5): error MSB4018:    at System.Net.WebClient.DownloadFile(Uri address, String fileName)
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(45,5): error MSB4018:    at System.Net.WebClient.DownloadFile(String address, String fileName)
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(45,5): error MSB4018:    at MSBuildTasks.GetMGCBTask.LocateMGCBFromNuget()
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(45,5): error MSB4018:    at MSBuildTasks.GetMGCBTask.Execute()
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(45,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(45,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()

**Comment by KevinYeti** [comment]
update 2： 
After modify the nuget proxy config file , I got a new error : 

error : failed to load “netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51” of type “System.Xml.Serialization.XmlRootAttribute”。
8>C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\build\MonoGame.Content.Builder.targets(73,5): error MSB3073: Command “C:\Users\T470\.nuget\packages\monogame.content.builder\3.7.0.9\tasks\net46\../../build\MGCB\build/MGCB.exe /@:"D:\MonoGame.Extended\Source\Demos\Platformer\Content\Content.mgcb" /platform:DesktopGL /quiet /outputDir:"bin\DesktopGL\Content" /intermediateDir:"obj\DesktopGL\Content"” exited，code 4。

**Comment by craftworkgames** [comment]
> failed to load “netstandard, Version=2.0.0.0

Did you install the [.NET Core SDK](https://dotnet.microsoft.com/download)?

**Comment by KevinYeti** [comment]
> > failed to load “netstandard, Version=2.0.0.0
> 
> Did you install the [.NET Core SDK](https://dotnet.microsoft.com/download)?

yes，  of course. I  surely installed the .NET Core SDK. 
But I have update the SDK to version 2.2 . 
```
Host (useful for support):
  Version: 2.2.0
  Commit:  1249f08fed

.NET Core SDKs installed:
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.402 [C:\Program Files\dotnet\sdk]
  2.1.403 [C:\Program Files\dotnet\sdk]
  2.1.500 [C:\Program Files\dotnet\sdk]
  2.1.502 [C:\Program Files\dotnet\sdk]
  2.1.503 [C:\Program Files\dotnet\sdk]
  2.2.100 [C:\Program Files\dotnet\sdk]
```

**Comment by craftworkgames** [comment]
I'm confused. Are you still having problems or not?

I don't know what the problem is. What I can tell from the error message is that the MonoGame Content Builder seems to be failing to load a .NET Standard library. 

Given that you're trying to build one of our demos and they build fine on many other machines out of the box I can only assume there's something about your machine that's different. 

We try to make it so that all of MonoGame.Extended's dependencies are downloaded and installed on the machine during the build as NuGet packages or using [Cake](https://cakebuild.net/).

Unfortunately, there's 3 dependencies that aren't available as versioned NuGet packages. They are the ones I listed earlier:
- Visual Studio 2017 (or some other compatible C# compiler),
- the MonoGame Content Builder 
- and the .NET Core SDK.

It's quite clear that the errors you're getting have something to do with these dependencies. I'm just not sure why. 

If I had a guess, I'd point the finger at the MonoGame Content Builder. It has proven to be problematic in the past because if your game runs a different version of MonoGame to the builder, it might not work.

**Comment by KevinYeti** [comment]
> I'm confused. Are you still having problems or not?
> 
> I don't know what the problem is. What I can tell from the error message is that the MonoGame Content Builder seems to be failing to load a .NET Standard library.
> 
> Given that you're trying to build one of our demos and they build fine on many other machines out of the box I can only assume there's something about your machine that's different.
> 
> We try to make it so that all of MonoGame.Extended's dependencies are downloaded and installed on the machine during the build as NuGet packages or using [Cake](https://cakebuild.net/).
> 
> Unfortunately, there's 3 dependencies that aren't available as versioned NuGet packages. They are the ones I listed earlier:
> 
> * Visual Studio 2017 (or some other compatible C# compiler),
> * the MonoGame Content Builder
> * and the .NET Core SDK.
> 
> It's quite clear that the errors you're getting have something to do with these dependencies. I'm just not sure why.
> 
> If I had a guess, I'd point the finger at the MonoGame Content Builder. It has proven to be problematic in the past because if your game runs a different version of MonoGame to the builder, it might not work.

Thank you for replying.

Still not working for me .  
I've tried different versions of MGCB tool but none get work. And the GUI of the MGCB tools not works for me either with the same error message. 

For now I'm turning to setup a new system and VS2017,  also a clean install of .NET Core SDK and the latest version of MonoGame to find whether it works.

**Comment by KevinYeti** [comment]
@craftworkgames 
I've something to update.

After I setup a new system and VS2017, also a clean install of .NET Core SDK, I tried to compile the project but still got a failure.

The error message says : 
`
Failed to load assembly 'C:/Users/T470/Desktop/MonoGame.Extended-release-3.6.0/Source/MonoGame.Extended.Content.Pipeline/bin/netstandard2.0/MonoGame.Extended.Content.Pipeline.dll'

C:/Users/T470/Desktop/MonoGame.Extended-release-3.6.0/Source/Demos/Platformer/Content/test-map.tmx: error: Failed to create importer 'TiledMapImporter'
`

I check the file [MonoGame.Extended.Content.Pipeline.dll] was there in the folder and all the files are at the right positon. Did I miss something ? 

BTW, I did not install the MonoGame Content Builder. Because I noticed the nuget will download a zip file named MGCB.zip and using it to build the content file. So I don't think that is necessary to install for the compiling.  Am I wrong ? Or should I install the latest version 3.7 of the MonoGame Content Builder ?

**Comment by craftworkgames** [comment]
When you got this error were you building with Visual Studio or `build.ps1`?

It's a bit of a long story but basically the `MonoGame.Extended.Content.Pipeline.dll` won't build in Visual Studio by default. This is due to the way dependencies are resolved for the selected project you're building. Since the Pipeline DLL is not technically a "real" dependency of the Platformer demo but instead it's a dependency of the MonoGame Content Builder it needs to be built separately. We've tried to resolve this problem before, but believe me when I say it's a tricky one.

Long story short, the `build.ps1` script will build things in the right order.

> I did not install the MonoGame Content Builder. Because I noticed the nuget will download a zip file named MGCB.zip and using it to build the content file. So I don't think that is necessary to install for the compiling. Am I wrong ?

Actually that's a good question. It's entirely possible that we solved this issue at some point. I know it was something we were trying to solve for quite a while. I guess if it works without installing the Builder then you're good to go.

**Comment by KevinYeti** [comment]
> When you got this error were you building with Visual Studio or `build.ps1`?

I 've tried `build.ps1` , but still got the same error message.

`Failed to load assembly 'C:/Users/T470/Desktop/mge/MonoGame.Extended/Source/MonoGame.Extended.Content.Pipeline/bin/netstandard2.0/MonoGame.Extended.Content.Pipeline.dll'`

**Comment by craftworkgames** [comment]
That is truly confusing. I don't know why it's happening.

I guess what I would do at this point is open up the `Content.mgcb` file in the MonoGame Content Builder tool and see if you can build the content that way.  If you get the same error at we'll know it's coming from the tool which might help narrow it down.

![image](https://user-images.githubusercontent.com/3201643/53711601-8b02b780-3e8e-11e9-83af-e4eac913299d.png)

Check that the "References" seem correct and are pointing at the right DLL.

![image](https://user-images.githubusercontent.com/3201643/53711636-be454680-3e8e-11e9-9be5-a6b4d1b08e45.png)

Check the version of the tool your using compared to mine maybe.

Some other things you could try is making a new MonoGame project and seeing if that works. Then add some content and see if that compiles and so on until you've manually rebuild the demo. 

Sorry, I'm really running out of ideas here.

**Comment by KevinYeti** [comment]
Still not working ~ 

![image](https://user-images.githubusercontent.com/615511/53713705-c8fbdd80-3e86-11e9-8ad4-b1c5606472de.png)
![image](https://user-images.githubusercontent.com/615511/53713727-d913bd00-3e86-11e9-9729-fddbf37f0b42.png)

But I noticed the Importer and Processor is Invalid. 
![image](https://user-images.githubusercontent.com/615511/53713772-06606b00-3e87-11e9-8c6b-e0b0332c74c9.png)

Althoungh I've checked many many times , the reference files are at the right place .

**Comment by craftworkgames** [comment]
@KevinYeti this looks a lot like issue #557 

You might want to have a read of the whole issue (it's not that long) but here's the important bits I think:

> **craftworkgames commented on Nov 28, 2018**
> I downloaded and installed the latest development build of MonoGame which happens to be MonoGame 3.8.0.173
> Repeating the same process I can confirm that this issue has been fixed in MonoGame and is not an issue in MonoGame.Extended.
> So there's essentially 2 workarounds to this problem.
> 1. You can install a development build of MonoGame 3.8.
> 2. If you need to continue using 3.7 for some reason, you can manually edit the Content.mgcb file to select the importer and processor.

> **mikeatuconn commented on Jan 21**
> I deleted the Content.mgcb file, opened the tool, made a new Content.mgcb file, added the reference, and everything worked.
